### PR TITLE
Add `/api/teams` endpoint

### PR DESCRIPTION
**Pull request by isamrish**
Signed-off-by: Amrish Kushwaha <askmaurya48@gmail.com>

This commit will add functionality that
/api/reviewers will list teams when no query parameter is passed

### CODE REVIEW

**Comment by itaditya** [comment]
@IsAmrish glad you picked this task, the logic seems right to me. I'll check it locally once I come back from home.
Things you need to change-

1. Remove the formatting you have done. I use prettier for formatting in this project. (single quote, trailing comma)
2. Instead of if-else, do an early return like this

```js
if(team.length === 0) {
  // team fetch logic
  return;
}

// old logic for when team length is not 0
```

**Comment by isamrish** [comment]
@itaditya  -- I have added .prettierrc file to format and refactored the code according to your suggestions.

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "trailingComma": "es5",
+  "singleQuote": true
```
Comment (itaditya) [inline .prettierrc]: add `printWidth` equal to `120` also and format accordingly

**Comment by itaditya** [comment]
@IsAmrish I think it would be better if you make a separate API for getting list of all teams in an ORG. Then on frontend based on query param provided or not you fetch this API and show dropdown.

It's great you abstracted the get teams logic in a function so it would be very easy to make the change.

Make this change and the prettier comment I have left and I'll merge the PR

**Comment by isamrish** [comment]
@itaditya  - separating API for teams is good idea. I think `teams` is good end point.
It also means I need to go back to previous code for the `reviewers` endpoint. right!!

**Comment by itaditya** [comment]
@IsAmrish There is one bug in your code. You are sending `name` of team. Actually name can be 'My Team' but slug would be `my-team` and every further GitHub API uses the team slug to get that team's data.

The `name` field can be used in the dropdown as the label. That means in dropdown user will see the list of names of teams. When they click on one item, we will find the slug corresponding to that item and then go ahead.

So basically we need both `name` and `slug` from API, so API res should be->

```js
{
  "teamsList": [
    {
      "name": "Some Team",
      "slug": "someteam"
    },
    {
      "name": "My Team",
      "slug": "my-team"
    }
  ]
}

```

**Comment by isamrish** [comment]
@itaditya -- thanks for pointing out. Fixing it.

**Comment by itaditya** [comment]
@IsAmrish thanks for working on it. Congrats your PR is merged!

Congrats
